common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt

```diff
@@ -151,13 +151,23 @@ class ContainerToolsRule(private val testInstance: Any) : TestRule {
      * Replaces the service binding in the [MutablePicoContainer] with the given instance and
      * returns the original service instance.
      *
+     * In case of using [mockk] for mocking, tries to extract original class name from a randomly
+     * assigned mocked class name.
+     *
      * @param newInstance the new instance to register
      */
     private fun setService(newInstance: Any) {
         with(ApplicationManager.getApplication().picoContainer as MutablePicoContainer) {
-            unregisterComponent(newInstance::class.java.name)
+            var javaClassName = newInstance::class.java.name
+            // when using Mockk, the class name for mocks is random and uses ByteBuddy suffix.
```

Comment:
interesting. I wonder why it was working previously for `@TestServices`s?

Reply:
also, what do you mean by the the name being random? presumably the prefix before the byteBuddy junk is always equal to the original classname?

Reply:
oh I also just had this thought (sorry for the barrage of comments):

Presumably mockito also had this problem with whatever bytecode manipulation framework it uses (cglib or whatever). I recall that mockito mocks' classname has a suffix as well (something like `...EnhancedByMockito...`). So I guess I'm wondering why this needs treatment that we didn't need in the Cloud Tools plugin?

Reply:
So the name of the mockk class is formed to something like this: `com.google.container.tools.skaffold.SkaffoldExecutorService$ByteBuddy$R5YLewgP$ByteBuddy$7xGX0mYJ`. Original class name is there, but it's appended with this ByteBuddy suffixes. Possibly Mockito has a different approach to this.

It only worked with `Spyk`, we haven't tried true `Mockk` with `TestService` before, so this came out.

Reply:
https://stackoverflow.com/questions/46191265/how-to-get-original-class-name-from-a-mocking-object 

That suggests that mockito does something similar. In any case, its not important if you've verified that it indeed needs this fix.

Reply:
Actually, I double checked what we did in Cloud Tools, and we used the original type of the field instead of the mocked object type. It should work here as well, I'll do the change.